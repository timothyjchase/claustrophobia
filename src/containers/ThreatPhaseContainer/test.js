import React from 'react'
import { ThreatPhaseContainer, mapStateToProps } from './index'

describe('<ThreatPhaseContainer />', () => {
  const commonProps = {
    completeThreatDemonPlacementStep: jest.fn(),
    completeThreatTrogsDistanceStep: jest.fn(),
    completeThreatTrogsPlacementStep: jest.fn(),
    completeThreatSpawnDemonStep: jest.fn(),
    completeThreatSpawnTrogsStep: jest.fn(),
    completeThreatEventStep: jest.fn(),
  }
  describe('with scenarioKey = TEST_SCENARIO', () => {
    const currentState = {
      scenarioKey: 'TEST_SCENARIO',
      threatStep: 'STEP',
      legalPlacement: true,
      threatDice: 6,
    }
    const component = shallow(
      <ThreatPhaseContainer {...commonProps} currentState={currentState} />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with scenarioKey = null', () => {
    const currentState = {
      scenarioKey: null,
      threatStep: 'STEP',
      legalPlacement: true,
      threatDice: 6,
    }
    const component = shallow(
      <ThreatPhaseContainer {...commonProps} currentState={currentState} />,
    )
    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})

describe('mapStateToProps', () => {
  describe('with basic state', () => {
    const state = { current: {} }
    it('should return currentState', () => {
      expect(mapStateToProps(state)).toEqual({ currentState: {} })
    })
  })
})
