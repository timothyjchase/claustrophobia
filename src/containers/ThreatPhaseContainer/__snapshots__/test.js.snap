// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ThreatPhaseContainer /> with scenarioKey = TEST_SCENARIO should match snapshot 1`] = `
<ThreatPhase
  completeThreatDemonPlacementStep={[MockFunction]}
  completeThreatEventStep={[MockFunction]}
  completeThreatSpawnDemonStep={[MockFunction]}
  completeThreatSpawnTrogsStep={[MockFunction]}
  completeThreatTrogsDistanceStep={[MockFunction]}
  completeThreatTrogsPlacementStep={[MockFunction]}
  demonName="Test Demon"
  legalPlacement={true}
  scenario={
    Object {
      "demon": "TEST_DEMON",
      "demonLimit": 1,
      "name": "Test Scenario",
    }
  }
  threatDice={6}
  threatStep="STEP"
/>
`;

exports[`<ThreatPhaseContainer /> with scenarioKey = null should match snapshot 1`] = `
<ThreatPhase
  completeThreatDemonPlacementStep={[MockFunction]}
  completeThreatEventStep={[MockFunction]}
  completeThreatSpawnDemonStep={[MockFunction]}
  completeThreatSpawnTrogsStep={[MockFunction]}
  completeThreatTrogsDistanceStep={[MockFunction]}
  completeThreatTrogsPlacementStep={[MockFunction]}
  demonName="Demon"
  legalPlacement={true}
  scenario={Object {}}
  threatDice={6}
  threatStep="STEP"
/>
`;
